Platon/backend/app/search_engine/views.py

```diff
@@ -308,7 +329,114 @@ def get(self):
+                            result_list.append({"id": upcoming_event[0], "title": upcoming_event[1], "acronym": upcoming_event[2],
+                                                "location": upcoming_event[3], "date": upcoming_event[4],
+                                                "deadline": upcoming_event[5], "link": upcoming_event[6]})
+                            result_id_score.append((upcoming_event[0], score))
+                        else:
+                            id_index = id_list.index(upcoming_event[0])
+                            result_id_score[id_index] = (upcoming_event[0], result_id_score[id_index][1] + score)
+                except:
+                    return make_response(jsonify({"error": "Database Connection Problem."}), 500)
 
+            # Search tokens for deadline match
+            deadline_filter_list = []
+            if form.deadline_filter.data != '':
+                for result in result_list:
+                    event_deadline = result.get("deadline")
+                    event_deadline = SearchEngine.remove_punctuation(event_deadline)
+                    remove_digits = str.maketrans('', '', digits)
+                    filter_data = form.deadline_filter.data.translate(remove_digits)
+                    filter_data = SearchEngine.remove_punctuation(filter_data)
+                    if filter_data in event_deadline:
+                        deadline_filter_list.append(result)
+                    result_list = deadline_filter_list
+            # Search tokens for date match
+            date_filter_list = []
+            if form.date_filter.data != '':
+                for result in result_list:
+                    event_date = result.get("date")
+                    event_date = SearchEngine.remove_punctuation(event_date)
+                    remove_digits = str.maketrans('', '', digits)
+                    filter_data = form.date_filter.data.translate(remove_digits)
+                    filter_data = SearchEngine.remove_punctuation(filter_data)
+                    if filter_data in event_date:
+                        date_filter_list.append(result)
+                    result_list = date_filter_list
+            sorted_result_list = []
+            # Sort result ids according to their scores
+            if form.sorting_criteria.data is not None:
+                # Sort Results according to Sorting Criteria
+                if form.sorting_criteria.data == 0:
+                    sorted_result_list = SearchEngine.sort_results(result_list, ["title"], False)
+                    result_list = sorted_result_list
+                else:
+                    dates = []
+                    for result in result_list:
+                        date = result.get("date")
+                        if date != "N/A":
+                            dates.append(date)
+
+                    dates.sort(key = lambda date: datetime.strptime(date[0:12].strip(), '%b %d, %Y'))
```

Comment:
Why didn't you use the part that I implemented on user search which sorts the results according to their relevance to the search query?